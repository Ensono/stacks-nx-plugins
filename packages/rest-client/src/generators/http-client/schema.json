{
    "$schema": "http://json-schema.org/schema",
    "cli": "nx",
    "$id": "HttpClient",
    "title": "Ensono Stacks Rest Client Http Client Generator",
    "description": "The @ensono-stacks/rest-client:http-client generator will add a rest client to an existing project",
    "type": "object",
    "properties": {
        "name": {
            "type": "string",
            "description": "Library name",
            "$default": {
                "$source": "argv",
                "index": 0
            },
            "x-prompt": "What name would you like to use for the library?",
            "pattern": "^[a-zA-Z0-9_-]+$"
        },
        "directory": {
            "type": "string",
            "description": "Subdirectory inside libs/ where the generated library placed.",
            "alias": "dir"
        },
        "importPath": {
            "type": "string",
            "description": "What import path would you like to use for the library?",
            "pattern": "^@[a-z0-9_/-]+$"
        },
        "tags": {
            "type": "string",
            "description": "Add tags to the library (used for linting).",
            "alias": "t"
        },
        "skipFormat": {
            "description": "Skip formatting files.",
            "type": "boolean",
            "default": false
        }
    },
    "required": ["name"]
}
