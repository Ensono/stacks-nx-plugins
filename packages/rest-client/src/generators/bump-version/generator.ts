import { normalizeOptions, copyFiles } from '@ensono-stacks/core';
import {
    formatFiles,
    names,
    readProjectConfiguration,
    Tree,
} from '@nrwl/devkit';
import { libraryGenerator } from '@nrwl/js';
import path from 'path';

import { BumpVersionGeneratorSchema } from './schema';

function findLatestVersion(
    tree: Tree,
    { directory, endpointName }: { directory?: string; endpointName: string },
): number {
    const versionsPath = readProjectConfiguration(
        tree,
        endpointName,
    ).root.replace(/v(\d+)$/g, '');

    const children = tree.children(versionsPath);

    if (children.length === 0) {
        throw new Error(
            "Could not find previous version of the endpoint. Are you sure you don't want to generate a new endpoint?",
        );
    }

    const versions = children.map(version =>
        Number.parseInt(version.replace(/^v/i, ''), 10),
    );

    return Math.max(...versions);
}

function determineNewVersion(
    latestVersion: number,
    newVersionParameter?: number | string,
): number {
    if (newVersionParameter !== undefined) {
        const newVersionNumber = Number(newVersionParameter);
        if (Number.isNaN(newVersionNumber)) {
            throw new TypeError('The endpoint version needs to be a number.');
        }
    }

    const newVersion = Number(newVersionParameter || latestVersion + 1);

    if (newVersion <= latestVersion) {
        throw new Error(
            `Cannot decrease a version. Please use --endpointVersion higher than ${latestVersion}`,
        );
    }

    return newVersion;
}

function updateVersionInCode(
    tree: Tree,
    filePaths: string[],
    latestVersion: number,
    latestVersionName: string,
    newVersion: number,
    newVersionName: string,
) {
    const latestVersionNames = names(latestVersionName);
    const newVersionNames = names(newVersionName);

    filePaths.forEach(filePath => {
        const fileContent = tree.read(filePath)?.toString();
        if (!fileContent) {
            return;
        }

        tree.write(
            filePath,
            fileContent
                .replace(
                    // eslint-disable-next-line security/detect-non-literal-regexp
                    new RegExp(latestVersionNames.className, 'g'),
                    newVersionNames.className,
                )
                .replace(
                    // eslint-disable-next-line security/detect-non-literal-regexp
                    new RegExp(latestVersionNames.name, 'g'),
                    newVersionNames.name,
                )
                .replace(
                    // eslint-disable-next-line security/detect-non-literal-regexp
                    new RegExp(`v${latestVersion}`, 'g'),
                    `v${newVersion}`,
                )
                .replace(
                    // eslint-disable-next-line security/detect-non-literal-regexp
                    new RegExp(`V${latestVersion}`, 'g'),
                    `V${newVersion}`,
                ),
        );
    });
}

export default async function bumpVersion(
    tree: Tree,
    optionsParameter: BumpVersionGeneratorSchema,
) {
    const latestVersion = findLatestVersion(tree, {
        directory: optionsParameter.directory,
        endpointName: optionsParameter.name,
    });

    if (Number.isNaN(latestVersion)) {
        throw new TypeError(
            `No version found. Please select your project with existing endpoints.`,
        );
    }

    const newVersion = determineNewVersion(
        latestVersion,
        optionsParameter.endpointVersion,
    );

    const libsEndpoint = readProjectConfiguration(
        tree,
        optionsParameter.name,
    ).root.replace(/v(\d+)$/g, '');

    const endpointRoot = libsEndpoint.replace('libs/', '');

    const latestVersionOptions = normalizeOptions(tree, {
        name: `${endpointRoot}v${latestVersion}`,
        directory: optionsParameter.directory,
        endpointName: endpointRoot,
        endpointVersion: latestVersion,
    });

    const newVersionOptions = normalizeOptions(tree, {
        ...optionsParameter,
        // include endpoint version in library name
        name: `${endpointRoot}v${newVersion}`,
        endpointName: endpointRoot,
        endpointVersion: newVersion,
    });

    // create a new library for the new version
    await libraryGenerator(tree, newVersionOptions);
    // Delete the default generated lib folder
    tree.delete(path.join(newVersionOptions.projectRoot, 'src', 'lib'));

    // copy files from latestVersion to newVersion
    copyFiles(
        tree,
        latestVersionOptions.projectRoot,
        newVersionOptions.projectRoot,
    );

    // update version numbers in the newly created files
    const filesToChange = tree
        .listChanges()
        .filter(change => change.path.startsWith(newVersionOptions.projectRoot))
        .map(change => change.path);
    updateVersionInCode(
        tree,
        filesToChange,
        latestVersion,
        latestVersionOptions.name,
        newVersion,
        newVersionOptions.name,
    );

    await formatFiles(tree);
}
