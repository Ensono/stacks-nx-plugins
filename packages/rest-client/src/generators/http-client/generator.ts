import {
    addCustomTestConfig,
    NormalizedSchema as BaseNormalizedSchema,
    normalizeOptions,
} from '@ensono-stacks/core';
import {
    addDependenciesToPackageJson,
    formatFiles,
    generateFiles,
    names,
    offsetFromRoot,
    readProjectConfiguration,
    Tree,
} from '@nrwl/devkit';
import { libraryGenerator } from '@nrwl/js';
import chalk from 'chalk';
import path from 'path';

import { axiosVersion } from '../../../utils/versions';
import { HttpClientGeneratorSchema } from './schema';

type NormalizedSchema = BaseNormalizedSchema<HttpClientGeneratorSchema>;

function updateDependencies(tree: Tree) {
    return addDependenciesToPackageJson(
        tree,
        {
            axios: axiosVersion,
        },
        {},
    );
}

function addFiles(tree: Tree, options: NormalizedSchema) {
    const templateOptions = {
        ...options,
        ...names(options.name),
        offsetFromRoot: offsetFromRoot(options.projectRoot),
        template: '',
    };

    generateFiles(
        tree,
        path.join(__dirname, 'files'),
        options.projectRoot,
        templateOptions,
    );
}

export default async function generate(
    tree: Tree,
    options: HttpClientGeneratorSchema,
) {
    const normalizedOptions = normalizeOptions(tree, options);

    // Use the existing library generator
    await libraryGenerator(tree, normalizedOptions);
    // Delete the default generated lib folder
    tree.delete(path.join(normalizedOptions.projectRoot, 'src', 'lib'));

    // Generate files
    addFiles(tree, normalizedOptions);
    // Update package.json
    updateDependencies(tree);

    const project = readProjectConfiguration(
        tree,
        normalizedOptions.projectName,
    );

    const ciCoverageConfig = {
        ci: {
            collectCoverage: true,
            coverageReporters: ['text', 'html'],
            collectCoverageFrom: [
                './**/*.{js,jsx,ts,tsx}',
                './!**/.next/**',
                './!**/*.d.ts',
                './!**/*.config.*',
                './!**/_app.*',
            ],
            codeCoverage: true,
            ci: true,
        },
    };

    await addCustomTestConfig(
        tree,
        project,
        normalizedOptions.projectName,
        ciCoverageConfig,
    );

    // Format files
    if (!normalizedOptions.skipFormat) {
        await formatFiles(tree);
    }

    if (normalizedOptions.directory) {
        console.log(
            chalk.yellow`NOTE: you generated the http client inside ${normalizedOptions.directory} directory, which means that the library is now called ${normalizedOptions.projectName}`,
        );
        console.log(
            chalk.yellow`      Remember this when running nx commands like "nx test ${normalizedOptions.projectName}"`,
        );
    }
}
