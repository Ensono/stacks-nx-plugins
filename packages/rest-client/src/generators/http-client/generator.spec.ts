import { tsMorphTree } from '@ensono-stacks/core';
import { checkFilesExistInTree } from '@ensono-stacks/test';
import {
    Tree,
    readProjectConfiguration,
    readJson,
    joinPathFragments,
} from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';

import generator from './generator';
import { HttpClientGeneratorSchema } from './schema';

function snapshotFiles(tree: Tree, files: string[]) {
    expect(() => checkFilesExistInTree(tree, ...files)).not.toThrowError();
    const project = tsMorphTree(tree);
    files.forEach(file => {
        expect(project.addSourceFileAtPath(file).getText()).toMatchSnapshot(
            file,
        );
    });
}

describe('http-client generator', () => {
    let tree: Tree;
    const options: HttpClientGeneratorSchema = {
        name: 'test-client',
        directory: 'test-client',
    };

    beforeEach(() => {
        tree = createTreeWithEmptyWorkspace();
    });

    it('should generate the http-client', async () => {
        await generator(tree, {
            ...options,
            tags: 'test, client',
        });
        const config = readProjectConfiguration(tree, 'test-client');
        expect(config).toBeDefined();
        expect(config.tags).toEqual(['test', 'client']);

        snapshotFiles(tree, [
            joinPathFragments(config.root, 'project.json'),
            joinPathFragments(config.root, 'tsconfig.json'),
            joinPathFragments(config.root, 'tsconfig.lib.json'),
            joinPathFragments(config.root, 'tsconfig.spec.json'),
            joinPathFragments(config.root, '.eslintrc.json'),
            joinPathFragments(config.root, 'jest.config.ts'),
            joinPathFragments(config.root, 'src', 'index.ts'),
            joinPathFragments(config.root, 'src', 'index.test.ts'),
            joinPathFragments(config.root, 'README.md'),
            'jest.config.ts',
            'jest.preset.js',
            '.prettierignore',
            '.prettierrc',
        ]);
    });

    it('should generate the http-client with custom directory', async () => {
        await generator(tree, {
            ...options,
            directory: 'custom',
            tags: 'test, client',
        });

        const config = readProjectConfiguration(tree, 'test-client');

        expect(config).toBeDefined();
        expect(config.tags).toEqual(['test', 'client']);

        snapshotFiles(tree, [
            joinPathFragments(config.root, 'project.json'),
            joinPathFragments(config.root, 'tsconfig.json'),
            joinPathFragments(config.root, 'tsconfig.lib.json'),
            joinPathFragments(config.root, 'tsconfig.spec.json'),
            joinPathFragments(config.root, '.eslintrc.json'),
            joinPathFragments(config.root, 'jest.config.ts'),
            joinPathFragments(config.root, 'src', 'index.ts'),
            joinPathFragments(config.root, 'src', 'index.test.ts'),
            joinPathFragments(config.root, 'README.md'),
            'jest.config.ts',
            'jest.preset.js',
            '.prettierignore',
            '.prettierrc',
        ]);
    });

    it('should generate the http-client with custom import path', async () => {
        await generator(tree, {
            ...options,
            importPath: '@customimport/restclient',
        });

        const config = readProjectConfiguration(tree, 'test-client');

        expect(config).toBeDefined();

        snapshotFiles(tree, [
            joinPathFragments(config.root, 'project.json'),
            joinPathFragments(config.root, 'tsconfig.json'),
            joinPathFragments(config.root, 'tsconfig.lib.json'),
            joinPathFragments(config.root, 'tsconfig.spec.json'),
            joinPathFragments(config.root, '.eslintrc.json'),
            joinPathFragments(config.root, 'jest.config.ts'),
            joinPathFragments(config.root, 'src', 'index.ts'),
            joinPathFragments(config.root, 'src', 'index.test.ts'),
            joinPathFragments(config.root, 'README.md'),
            'jest.config.ts',
            'jest.preset.js',
            '.prettierignore',
            '.prettierrc',
        ]);

        const tsConfig = readJson(tree, `tsconfig.base.json`);
        expect(tsConfig.compilerOptions.paths).toHaveProperty(
            '@customimport/restclient',
        );
    });

    it('should generate the http-client with no formatting', async () => {
        await generator(tree, {
            ...options,
            skipFormat: true,
        });

        const config = readProjectConfiguration(tree, 'test-client');

        expect(config).toBeDefined();

        snapshotFiles(tree, [
            joinPathFragments(config.root, 'project.json'),
            joinPathFragments(config.root, 'tsconfig.json'),
            joinPathFragments(config.root, 'tsconfig.lib.json'),
            joinPathFragments(config.root, 'tsconfig.spec.json'),
            joinPathFragments(config.root, '.eslintrc.json'),
            joinPathFragments(config.root, 'jest.config.ts'),
            joinPathFragments(config.root, 'src', 'index.ts'),
            joinPathFragments(config.root, 'src', 'index.test.ts'),
            joinPathFragments(config.root, 'README.md'),
            'jest.config.ts',
            'jest.preset.js',
            '.prettierignore',
            '.prettierrc',
        ]);
    });

    it('should install axios as dependency', async () => {
        await generator(tree, options);

        const packageJson = readJson(tree, 'package.json');
        expect(Object.keys(packageJson.dependencies)).toEqual(
            expect.arrayContaining(['axios']),
        );
    });
});
