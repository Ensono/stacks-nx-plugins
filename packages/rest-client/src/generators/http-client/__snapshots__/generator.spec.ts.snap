// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`http-client generator should generate the http-client with custom directory: .prettierignore 1`] = `
"# Add files here to ignore them from prettier formatting
/dist
/coverage
/.nx/cache"
`;

exports[`http-client generator should generate the http-client with custom directory: .prettierrc 1`] = `
"{ "singleQuote": true }
"
`;

exports[`http-client generator should generate the http-client with custom directory: custom/test-client/.eslintrc.json 1`] = `
"{
  "extends": ["../../.eslintrc.json"],
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "rules": {}
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {}
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    }
  ]
}
"
`;

exports[`http-client generator should generate the http-client with custom directory: custom/test-client/README.md 1`] = `
"# custom-test-client

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run \`nx test custom-test-client\` to execute the unit tests via [Jest](https://jestjs.io).
"
`;

exports[`http-client generator should generate the http-client with custom directory: custom/test-client/jest.config.ts 1`] = `
"export default {
  displayName: 'custom-test-client',
  preset: '../../jest.preset.js',
  testEnvironment: 'node',
  transform: {
    '^.+\\\\.[tj]s$': ['ts-jest', { tsconfig: '<rootDir>/tsconfig.spec.json' }],
  },
  moduleFileExtensions: ['ts', 'js', 'html'],
  coverageDirectory: '../../coverage/custom/test-client',
};
"
`;

exports[`http-client generator should generate the http-client with custom directory: custom/test-client/project.json 1`] = `
"{
  "name": "custom-test-client",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "custom/test-client/src",
  "projectType": "library",
  "tags": ["test", "client"],
  "targets": {
    "lint": {
      "executor": "@nx/eslint:lint"
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "custom/test-client/jest.config.ts"
      },
      "configurations": {
        "ci": {
          "collectCoverage": true,
          "coverageReporters": ["text", "html"],
          "collectCoverageFrom": [
            "./**/*.{js,jsx,ts,tsx}",
            "./!**/.next/**",
            "./!**/*.d.ts",
            "./!**/*.config.*",
            "./!**/_app.*"
          ],
          "codeCoverage": true,
          "ci": true
        }
      }
    }
  }
}
"
`;

exports[`http-client generator should generate the http-client with custom directory: custom/test-client/src/index.test.ts 1`] = `
"import httpClient from '.';

describe('axios', () => {
  it('should be an axios instance', () => {
    expect(httpClient).toBeDefined();
    expect(httpClient).toHaveProperty('get');
    expect(httpClient).toHaveProperty('post');
  });
});
"
`;

exports[`http-client generator should generate the http-client with custom directory: custom/test-client/src/index.ts 1`] = `
"import axios, { AxiosRequestConfig, AxiosResponse } from 'axios';

// Reusable instance
const httpClient = axios.create();

export type RequestConfig<T = any> = AxiosRequestConfig<T>;
export type Response<T = any> = AxiosResponse<T>;

export default httpClient;
"
`;

exports[`http-client generator should generate the http-client with custom directory: custom/test-client/tsconfig.json 1`] = `
"{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "module": "commonjs"
  },
  "files": [],
  "include": [],
  "references": [
    {
      "path": "./tsconfig.lib.json"
    },
    {
      "path": "./tsconfig.spec.json"
    }
  ]
}
"
`;

exports[`http-client generator should generate the http-client with custom directory: custom/test-client/tsconfig.lib.json 1`] = `
"{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "outDir": "../../dist/out-tsc",
    "declaration": true,
    "types": ["node"]
  },
  "include": ["src/**/*.ts"],
  "exclude": ["jest.config.ts", "src/**/*.spec.ts", "src/**/*.test.ts"]
}
"
`;

exports[`http-client generator should generate the http-client with custom directory: custom/test-client/tsconfig.spec.json 1`] = `
"{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "outDir": "../../dist/out-tsc",
    "module": "commonjs",
    "types": ["jest", "node"]
  },
  "include": [
    "jest.config.ts",
    "src/**/*.test.ts",
    "src/**/*.spec.ts",
    "src/**/*.d.ts"
  ]
}
"
`;

exports[`http-client generator should generate the http-client with custom directory: jest.config.ts 1`] = `
"import { getJestProjectsAsync } from '@nx/jest';

export default async () => ({
  projects: await getJestProjectsAsync(),
});
"
`;

exports[`http-client generator should generate the http-client with custom directory: jest.preset.js 1`] = `
"const nxPreset = require('@nx/jest/preset').default;

module.exports = { ...nxPreset };
"
`;

exports[`http-client generator should generate the http-client with custom import path: .prettierignore 1`] = `
"# Add files here to ignore them from prettier formatting
/dist
/coverage
/.nx/cache"
`;

exports[`http-client generator should generate the http-client with custom import path: .prettierrc 1`] = `
"{ "singleQuote": true }
"
`;

exports[`http-client generator should generate the http-client with custom import path: jest.config.ts 1`] = `
"import { getJestProjectsAsync } from '@nx/jest';

export default async () => ({
  projects: await getJestProjectsAsync(),
});
"
`;

exports[`http-client generator should generate the http-client with custom import path: jest.preset.js 1`] = `
"const nxPreset = require('@nx/jest/preset').default;

module.exports = { ...nxPreset };
"
`;

exports[`http-client generator should generate the http-client with custom import path: test-client/.eslintrc.json 1`] = `
"{
  "extends": ["../.eslintrc.json"],
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "rules": {}
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {}
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    }
  ]
}
"
`;

exports[`http-client generator should generate the http-client with custom import path: test-client/README.md 1`] = `
"# test-client

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run \`nx test test-client\` to execute the unit tests via [Jest](https://jestjs.io).
"
`;

exports[`http-client generator should generate the http-client with custom import path: test-client/jest.config.ts 1`] = `
"export default {
  displayName: 'test-client',
  preset: '../jest.preset.js',
  testEnvironment: 'node',
  transform: {
    '^.+\\\\.[tj]s$': ['ts-jest', { tsconfig: '<rootDir>/tsconfig.spec.json' }],
  },
  moduleFileExtensions: ['ts', 'js', 'html'],
  coverageDirectory: '../coverage/test-client',
};
"
`;

exports[`http-client generator should generate the http-client with custom import path: test-client/project.json 1`] = `
"{
  "name": "test-client",
  "$schema": "../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "test-client/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "lint": {
      "executor": "@nx/eslint:lint"
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "test-client/jest.config.ts"
      },
      "configurations": {
        "ci": {
          "collectCoverage": true,
          "coverageReporters": ["text", "html"],
          "collectCoverageFrom": [
            "./**/*.{js,jsx,ts,tsx}",
            "./!**/.next/**",
            "./!**/*.d.ts",
            "./!**/*.config.*",
            "./!**/_app.*"
          ],
          "codeCoverage": true,
          "ci": true
        }
      }
    }
  }
}
"
`;

exports[`http-client generator should generate the http-client with custom import path: test-client/src/index.test.ts 1`] = `
"import httpClient from '.';

describe('axios', () => {
  it('should be an axios instance', () => {
    expect(httpClient).toBeDefined();
    expect(httpClient).toHaveProperty('get');
    expect(httpClient).toHaveProperty('post');
  });
});
"
`;

exports[`http-client generator should generate the http-client with custom import path: test-client/src/index.ts 1`] = `
"import axios, { AxiosRequestConfig, AxiosResponse } from 'axios';

// Reusable instance
const httpClient = axios.create();

export type RequestConfig<T = any> = AxiosRequestConfig<T>;
export type Response<T = any> = AxiosResponse<T>;

export default httpClient;
"
`;

exports[`http-client generator should generate the http-client with custom import path: test-client/tsconfig.json 1`] = `
"{
  "extends": "../tsconfig.base.json",
  "compilerOptions": {
    "module": "commonjs"
  },
  "files": [],
  "include": [],
  "references": [
    {
      "path": "./tsconfig.lib.json"
    },
    {
      "path": "./tsconfig.spec.json"
    }
  ]
}
"
`;

exports[`http-client generator should generate the http-client with custom import path: test-client/tsconfig.lib.json 1`] = `
"{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "outDir": "../dist/out-tsc",
    "declaration": true,
    "types": ["node"]
  },
  "include": ["src/**/*.ts"],
  "exclude": ["jest.config.ts", "src/**/*.spec.ts", "src/**/*.test.ts"]
}
"
`;

exports[`http-client generator should generate the http-client with custom import path: test-client/tsconfig.spec.json 1`] = `
"{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "outDir": "../dist/out-tsc",
    "module": "commonjs",
    "types": ["jest", "node"]
  },
  "include": [
    "jest.config.ts",
    "src/**/*.test.ts",
    "src/**/*.spec.ts",
    "src/**/*.d.ts"
  ]
}
"
`;

exports[`http-client generator should generate the http-client with no formatting: .prettierignore 1`] = `
"# Add files here to ignore them from prettier formatting
/dist
/coverage
/.nx/cache"
`;

exports[`http-client generator should generate the http-client with no formatting: .prettierrc 1`] = `"{"singleQuote":true}"`;

exports[`http-client generator should generate the http-client with no formatting: jest.config.ts 1`] = `
"import { getJestProjectsAsync } from '@nx/jest';

export default async () => ({
projects: await getJestProjectsAsync()
});"
`;

exports[`http-client generator should generate the http-client with no formatting: jest.preset.js 1`] = `
"const nxPreset = require('@nx/jest/preset').default;

      module.exports = { ...nxPreset }"
`;

exports[`http-client generator should generate the http-client with no formatting: test-client/.eslintrc.json 1`] = `
"{
  "extends": [
    "../.eslintrc.json"
  ],
  "ignorePatterns": [
    "!**/*"
  ],
  "overrides": [
    {
      "files": [
        "*.ts",
        "*.tsx",
        "*.js",
        "*.jsx"
      ],
      "rules": {}
    },
    {
      "files": [
        "*.ts",
        "*.tsx"
      ],
      "rules": {}
    },
    {
      "files": [
        "*.js",
        "*.jsx"
      ],
      "rules": {}
    }
  ]
}
"
`;

exports[`http-client generator should generate the http-client with no formatting: test-client/README.md 1`] = `
"# test-client

This library was generated with [Nx](https://nx.dev).





## Running unit tests

Run \`nx test test-client\` to execute the unit tests via [Jest](https://jestjs.io).


"
`;

exports[`http-client generator should generate the http-client with no formatting: test-client/jest.config.ts 1`] = `
"export default {
  displayName: 'test-client',
  preset: '../jest.preset.js',
  testEnvironment: 'node',
  transform: {
    '^.+\\\\.[tj]s$': ['ts-jest', { tsconfig: '<rootDir>/tsconfig.spec.json' }]
  },
  moduleFileExtensions: ['ts', 'js', 'html'],
  coverageDirectory: '../coverage/test-client'
};
"
`;

exports[`http-client generator should generate the http-client with no formatting: test-client/project.json 1`] = `
"{
  "name": "test-client",
  "$schema": "../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "test-client/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "lint": {
      "executor": "@nx/eslint:lint"
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": [
        "{workspaceRoot}/coverage/{projectRoot}"
      ],
      "options": {
        "jestConfig": "test-client/jest.config.ts"
      },
      "configurations": {
        "ci": {
          "collectCoverage": true,
          "coverageReporters": [
            "text",
            "html"
          ],
          "collectCoverageFrom": [
            "./**/*.{js,jsx,ts,tsx}",
            "./!**/.next/**",
            "./!**/*.d.ts",
            "./!**/*.config.*",
            "./!**/_app.*"
          ],
          "codeCoverage": true,
          "ci": true
        }
      }
    }
  }
}
"
`;

exports[`http-client generator should generate the http-client with no formatting: test-client/src/index.test.ts 1`] = `
"import httpClient from '.';

describe('axios', () => {
  it('should be an axios instance', () => {
    expect(httpClient).toBeDefined();
    expect(httpClient).toHaveProperty('get');
    expect(httpClient).toHaveProperty('post');
  });
});"
`;

exports[`http-client generator should generate the http-client with no formatting: test-client/src/index.ts 1`] = `
"import axios, { AxiosRequestConfig, AxiosResponse } from 'axios';

// Reusable instance
const httpClient = axios.create();

export type RequestConfig<T = any> = AxiosRequestConfig<T>;
export type Response<T = any> = AxiosResponse<T>;

export default httpClient;
"
`;

exports[`http-client generator should generate the http-client with no formatting: test-client/tsconfig.json 1`] = `
"{
  "extends": "../tsconfig.base.json",
  "compilerOptions": {
    "module": "commonjs"
  },
  "files": [],
  "include": [],
  "references": [
    {
      "path": "./tsconfig.lib.json"
    },
    {
      "path": "./tsconfig.spec.json"
    }
  ]
}
"
`;

exports[`http-client generator should generate the http-client with no formatting: test-client/tsconfig.lib.json 1`] = `
"{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "outDir": "../dist/out-tsc",
    "declaration": true,
    "types": ["node"]
  },
  "include": ["src/**/*.ts"],
  "exclude": [
    "jest.config.ts", 
    "src/**/*.spec.ts", "src/**/*.test.ts"
  ]
}
"
`;

exports[`http-client generator should generate the http-client with no formatting: test-client/tsconfig.spec.json 1`] = `
"{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "outDir": "../dist/out-tsc",
    "module": "commonjs",
    "types": ["jest", "node"]
  },
  "include": [
    "jest.config.ts",
    "src/**/*.test.ts",
    "src/**/*.spec.ts",
    "src/**/*.d.ts"
  ]
}
"
`;

exports[`http-client generator should generate the http-client: .prettierignore 1`] = `
"# Add files here to ignore them from prettier formatting
/dist
/coverage
/.nx/cache"
`;

exports[`http-client generator should generate the http-client: .prettierrc 1`] = `
"{ "singleQuote": true }
"
`;

exports[`http-client generator should generate the http-client: jest.config.ts 1`] = `
"import { getJestProjectsAsync } from '@nx/jest';

export default async () => ({
  projects: await getJestProjectsAsync(),
});
"
`;

exports[`http-client generator should generate the http-client: jest.preset.js 1`] = `
"const nxPreset = require('@nx/jest/preset').default;

module.exports = { ...nxPreset };
"
`;

exports[`http-client generator should generate the http-client: test-client/.eslintrc.json 1`] = `
"{
  "extends": ["../.eslintrc.json"],
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "rules": {}
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {}
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    }
  ]
}
"
`;

exports[`http-client generator should generate the http-client: test-client/README.md 1`] = `
"# test-client

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run \`nx test test-client\` to execute the unit tests via [Jest](https://jestjs.io).
"
`;

exports[`http-client generator should generate the http-client: test-client/jest.config.ts 1`] = `
"export default {
  displayName: 'test-client',
  preset: '../jest.preset.js',
  testEnvironment: 'node',
  transform: {
    '^.+\\\\.[tj]s$': ['ts-jest', { tsconfig: '<rootDir>/tsconfig.spec.json' }],
  },
  moduleFileExtensions: ['ts', 'js', 'html'],
  coverageDirectory: '../coverage/test-client',
};
"
`;

exports[`http-client generator should generate the http-client: test-client/project.json 1`] = `
"{
  "name": "test-client",
  "$schema": "../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "test-client/src",
  "projectType": "library",
  "tags": ["test", "client"],
  "targets": {
    "lint": {
      "executor": "@nx/eslint:lint"
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "test-client/jest.config.ts"
      },
      "configurations": {
        "ci": {
          "collectCoverage": true,
          "coverageReporters": ["text", "html"],
          "collectCoverageFrom": [
            "./**/*.{js,jsx,ts,tsx}",
            "./!**/.next/**",
            "./!**/*.d.ts",
            "./!**/*.config.*",
            "./!**/_app.*"
          ],
          "codeCoverage": true,
          "ci": true
        }
      }
    }
  }
}
"
`;

exports[`http-client generator should generate the http-client: test-client/src/index.test.ts 1`] = `
"import httpClient from '.';

describe('axios', () => {
  it('should be an axios instance', () => {
    expect(httpClient).toBeDefined();
    expect(httpClient).toHaveProperty('get');
    expect(httpClient).toHaveProperty('post');
  });
});
"
`;

exports[`http-client generator should generate the http-client: test-client/src/index.ts 1`] = `
"import axios, { AxiosRequestConfig, AxiosResponse } from 'axios';

// Reusable instance
const httpClient = axios.create();

export type RequestConfig<T = any> = AxiosRequestConfig<T>;
export type Response<T = any> = AxiosResponse<T>;

export default httpClient;
"
`;

exports[`http-client generator should generate the http-client: test-client/tsconfig.json 1`] = `
"{
  "extends": "../tsconfig.base.json",
  "compilerOptions": {
    "module": "commonjs"
  },
  "files": [],
  "include": [],
  "references": [
    {
      "path": "./tsconfig.lib.json"
    },
    {
      "path": "./tsconfig.spec.json"
    }
  ]
}
"
`;

exports[`http-client generator should generate the http-client: test-client/tsconfig.lib.json 1`] = `
"{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "outDir": "../dist/out-tsc",
    "declaration": true,
    "types": ["node"]
  },
  "include": ["src/**/*.ts"],
  "exclude": ["jest.config.ts", "src/**/*.spec.ts", "src/**/*.test.ts"]
}
"
`;

exports[`http-client generator should generate the http-client: test-client/tsconfig.spec.json 1`] = `
"{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "outDir": "../dist/out-tsc",
    "module": "commonjs",
    "types": ["jest", "node"]
  },
  "include": [
    "jest.config.ts",
    "src/**/*.test.ts",
    "src/**/*.spec.ts",
    "src/**/*.d.ts"
  ]
}
"
`;
